### DISCUSSION TITLE
我想把axum::middleware::from_fn 这个函数作为另一个函数的返回值返回

### DISCUSSION THREAD

**Discussion by igufei** [tokio-rs/axum#1886] [Q&A] [answered]
#### I want to return the function axum::middleware::from_fn as the return value of another function
```
fn handler<F,T>(jwt_secret: &str) -> FromFnLayer<F, (), T>
{
    from_fn(|request: Request<T>, next: Next<T>| async move {
        next.run(request).await
    });
}
```
> 上面的代码提示`mismatched types`
我应该怎么修复
我是个axum新手

**Comment by davidpdrsn** [comment, marked as answer]
You can't easily do that. `async {}` returns an unnameable future type. I'd recommend using https://docs.rs/axum/latest/axum/middleware/fn.from_fn_with_state.html instead.

**Reply by igufei** [reply]
`from_fn(move |req, next| auth(JWT_SECRET, req, next))`
看来我只能用这种方式了,谢谢你的回答
